Source/Adapters/Blackjack.Adapters/UserAdapter.cs

```diff
@@ -54,6 +62,19 @@ public async Task<UserBE> GetUsers(string username)
         return usersBySoundex?.SingleOrDefault(user => user.Username.Equals(username, StringComparison.OrdinalIgnoreCase));
     }
 
+    public async Task<UserBE> UpdateUser(UserBE user)
+    {
+        UserBE dbUser = await GetUsers(user.Username);
+        if (dbUser == null)
+        {
+            throw new WebException(HttpStatusCode.BadRequest, "User does not exist.");
+        }
+        dbUser.Username = user.Username;
+        dbUser.Balance = user.Balance;
+        List<UserBE> updatedUsers = await _userFacade.UpdateUser(dbUser);
```

Comment:
You're working with a single user, this facade.UpdateUser has no reason to return a list of users, it should never work with more than one user.

Reply:
Well I thought this would be in a similar vein of what we did with GetUsers. Since the initial fetch from the database has to return a list of users, even if its just a single user, you had me move the single or default to the adapter layer since it could be considered business logic. I thought I would do something like that here. I'll change it but wanted to double check

Reply:
Update as a whole, like as a web service endpoint, should return the user as they exist in the database at the end of everything.

But in this case the database does not generate any data that you do not already have in code provided via the UI.

So this UserUpdate workflow has no reason to return anything from the repo layer, and only needs to return the user it was given from the facade layer after the update occurs.

Reply:
Ok that makes sense. In that case, how far up can I stop returning something since I already have the data that will be in the DB? Could I just make only the controller return something? Or at least go as deep as the adapter where the conversions are occurring?

Reply:
This should become apparent when you make the change. See if you can determine it for yourself first.